libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -309,12 +304,9 @@ void AP_Terrain::update(void)
     // just schedule any needed disk IO
     schedule_disk_io();
 
-    // try to ensure the home location is populated
-    float height;
-    height_amsl(ahrs.get_home(), height, false);
```

Comment:
@WickedShell Besides the case about changing home you've pointed out, I think you are missing a case where terrain data might not be available when home is set/on takeoff - which is the case I thought about when dismissing this in Coverity.

The reason for having a special case for home is that it is always used for calculating the current altitude above terrain. I think this library could be changed to not depend on home altitude as it already depends on having the current location - and as a consequence we also have the current absolute altitude, meaning we don't need to calculate terrain difference with home.

Concluding: I think it's better to leave this in.